knora/dsplib/utils/xml_upload.py

```diff
@@ -267,17 +243,15 @@ def xml_upload(input_file: str, server: str, user: str, password: str, imgdir: s
     # Validate the input XML file
     current_dir = os.path.dirname(os.path.realpath(__file__))
     schema_file = os.path.join(current_dir, '../schemas/data.xsd')
-    if _validate_xml_against_schema(input_file, schema_file):
-        print("The input data file is syntactically correct and passed validation.")
-        if validate_only:
-            exit(0)
-    else:
-        print("ERROR The input data file did not pass validation.")
-        exit(1)
+    try:
+        validate_xml_against_schema(input_file, schema_file)
+    except BaseError as err:
+        print(err)
+        quit(0)
```

Comment:
@Irina: Here I don't get the effect that I'd like to: I thought it would be a good design pattern to raise an error if the schema validation fails, and then to catch the error and print out the details. Plus, I want to hide the stacktrace from the user if validation fails. But with the code that I have now, I get the following output when the XML is invalid:

```
dsp-tools xmlupload testdata/test-data-systematic.xml --validate
Traceback (most recent call last):
  File "/Users/nusjoh00-adm/.local/share/virtualenvs/dsp-tools--UOnuDrn/bin/dsp-tools", line 8, in <module>
    sys.exit(main())
  File "/Users/nusjoh00-adm/Desktop/dsp-tools/knora/dsp_tools.py", line 215, in main
    program(sys.argv[1:])
  File "/Users/nusjoh00-adm/Desktop/dsp-tools/knora/dsp_tools.py", line 181, in program
    validate_xml_against_schema(input_file=args.xmlfile,
  File "/Users/nusjoh00-adm/Desktop/dsp-tools/knora/dsplib/utils/shared_methods.py", line 109, in validate_xml_against_schema
    raise BaseError(error_msg)
knora.dsplib.models.helpers.BaseError: ERROR: The input data file cannot be uploaded due to the following validation error(s):
  Line 386: Element '{https://dasch.swiss/schema}resource', attribute 'labell': The attribute 'labell' is not allowed.
  Line 386: Element '{https://dasch.swiss/schema}resource': The attribute 'label' is required but missing.
  Line 402: Element '{https://dasch.swiss/schema}text', attribute 'encoding': [facet 'enumeration'] The value 'öalkj' is not an element of the set {'utf8', 'xml'}.!

NoneType: None
```

I don't know why the error escalates into the console so that the user sees it. I thought that if I catch it, it is "neutralized".

Can you give me advice how to improve this? (Or if I can just leave it as it is?)

Reply:
Hm, interesting. I have no idea why it doesn't work. Did it work in other locations?

Reply:
Actually, it's not worth the effort. I can just as well let the error escalate. The user will see the stack trace, but that doesn't hurt.

Reply:
I agree regarding the user experience. It's not too bad. But I worry about the source of the problem - this could mean that something "deeper" is wrong here. But I leave it up to you, of course, if you want to track this down or not.

Reply:
Thanks for the incentive to investigate. It was worth it: I found out that I had made a stupid mistake: If I print the entire `err` object, it's only logical that the stacktrace is printed. I have to print err.message instead :-)